### ISSUE TITLE
Automatic building on turn 2

### ISSUE THREAD

**Issue by jpeterbaker** [jpeterbaker/bga_homeworlds#4]
To save time in turn-based games, I suggest an option to automatically build a ship on turn 2 without waiting for player input.

**Comment by SimonN** [comment]
Yes. I believe that this second-turn build is always correct as long as the build is possible (homeworld has green) for the first player. Not even second player placing his homeworld adjacent to ours, regardless of colors, would deter us from the second-turn build.

There are abstruse scenarios where second player shouldn't build, but they're a loss for player 2 regardless of what player 2 does; not building would merely delay the loss. Example:

1. Player 1 Homeworld green-yellow, r3 ship
1... Player 2 Homeworld adjacent, but without red (blunder)
2. Player 1 builds small red.
2... Player 2 changes the ship to red to defend against immediate invasion, then passes on all future turns.

Thus, even in this case, always building in turn 2 is correct in the sense that it cannot worsen the game outcome.

Does BGA support conditional moves? I doubt it because "move" means vastly different things in different BGA games.

I can imagine only one reason for not automating second-turn build altogether: The newbie would be surprised.

**Comment by th555** [comment]
What about possibility that p1 wants to change his strategy in response to p2's homeworld/ship choice, by exchanging his large for a different colour? Sure you might argue that this is suboptimal, but one could have prepared certain strategies etc...

As long as it remains optional, I would be totally fine with it, though.
